CRM/Contact/BAO/Relationship.php

```diff
@@ -620,7 +620,10 @@ public static function clearCurrentEmployer($id, $action) {
         $sharedContact->id = $relationship->contact_id_a;
         $sharedContact->find(TRUE);
 
-        if ($relationship->relationship_type_id == 4 && $relationship->contact_id_b == $sharedContact->employer_id) {
+        // CRM-15881 UPDATES
+        // changed FROM "...relationship->relationship_type_id == 4..." TO "...relationship->relationship_type_id == 5..."
+        // As the system should be looking for type "employer of" (id 5) and not "sibling of" (id 4)
+        if ($relationship->relationship_type_id == 5 && $relationship->contact_id_b == $sharedContact->employer_id) {
```

Comment:
This bug is caused by the fact that relationship_type.id values may be different for different sites. Let's not use the constant '5', but instead get the Employee/Employer relationship type using the name column (which is not translated / edited):
$empRelTypeId = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', 'Employee of', 'id', 'name_a_b');